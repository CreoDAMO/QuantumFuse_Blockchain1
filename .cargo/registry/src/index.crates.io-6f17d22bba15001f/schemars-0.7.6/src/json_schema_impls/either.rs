use crate::gen::SchemaGenerator;
use crate::schema::*;
use crate::JsonSchema;
use either::Either;

impl<L: JsonSchema, R: JsonSchema> JsonSchema for Either<L, R> {
    no_ref_schema!();

    fn schema_name() -> String {
        format!("Either_{}_or_{}", L::schema_name(), R::schema_name())
    }

    fn json_schema(gen: &mut SchemaGenerator) -> Schema {
        let mut schema = SchemaObject::default();
        schema.subschemas().any_of = Some(vec![gen.subschema_for::<L>(), gen.subschema_for::<R>()]);
        schema.into()
    }
}
