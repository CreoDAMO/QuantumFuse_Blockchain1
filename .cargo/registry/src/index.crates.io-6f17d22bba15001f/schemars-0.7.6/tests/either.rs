mod util;
use either::Either;
use util::*;

#[test]
fn either() -> TestResult {
    test_default_generated_schema::<Either<i32, Either<bool, ()>>>("either")
}
