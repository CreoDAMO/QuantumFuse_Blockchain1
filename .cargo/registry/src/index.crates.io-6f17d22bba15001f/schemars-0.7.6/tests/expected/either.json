{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Either_int32_or_Either_Boolean_or_Null",
  "anyOf": [
    {
      "type": "integer",
      "format": "int32"
    },
    {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "type": "null"
        }
      ]
    }
  ]
}